### PR TITLE
My hw 03

### PR DESCRIPTION
**Pull request by Julittaes**

### CODE REVIEW

**Comment by vladhilko** [comment]
CI is red

**Comment by Julittaes** [comment]
Could you tell me is it possible to run tests somehow by myself not waiting for approval?

**Comment by vladhilko** [comment]
> Could you tell me is it possible to run tests somehow by myself not waiting for approval?

You can just run the following inside homework folder:

```rb
bundle install
rspec 
```

**Comment by Julittaes** [comment]
Fixed code and checked via rubocop and rspec locally, everything should work now, could you please approve tests run

File: HW03/yuliya_sheremet/yuliya_sheremet_hw_03_t_03.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  def check_and_print(seconds)
+    if seconds.size <= 2
+      print_a(seconds)
+    elsif seconds.size > 2
+      print_b(seconds)
+    end
+  end
+
+  def if_include(array_of_strings)
```
Comment (vladhilko) [inline HW03/yuliya_sheremet/yuliya_sheremet_hw_03_t_03.rb:42]: We don't use names with `if`, we can use something like this
```suggestion
  def include_calling_core?(array_of_strings)
```

File: HW03/yuliya_sheremet/yuliya_sheremet_hw_03_t_03.rb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    array_of_strings.select! { |line| line.include? 'Calling core' }
+    array_of_strings.map! do |p|
+      data = p.split(' ')
+      [data[0], data[1]].join(' ')
+    end
```
Comment (vladhilko) [inline HW03/yuliya_sheremet/yuliya_sheremet_hw_03_t_03.rb:47]: No need to use temp variable here
```suggestion
    array_of_strings.select { |line| line.include? 'Calling core' }
       .map do |p|
      data = p.split(' ')
      [data[0], data[1]].join(' ')
    end
```

File: HW03/yuliya_sheremet/yuliya_sheremet_hw_03_t_03.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    if seconds.size <= 2
+      print_a(seconds)
+    elsif seconds.size > 2
+      print_b(seconds)
+    end
+  end
+
+  def if_include(array_of_strings)
+    array_of_strings.select! { |line| line.include? 'Calling core' }
+    array_of_strings.map! do |p|
```
Comment (vladhilko) [inline HW03/yuliya_sheremet/yuliya_sheremet_hw_03_t_03.rb:44]: can we use a clear name for the variable?
```suggestion
    array_of_strings.map! do |line|
```

File: HW03/yuliya_sheremet/yuliya_sheremet_hw_03_t_03.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+      data = p.split(' ')
+      [data[0], data[1]].join(' ')
```
Comment (vladhilko) [inline HW03/yuliya_sheremet/yuliya_sheremet_hw_03_t_03.rb:46]: Something like this should work 
```suggestion
      p.split(' ')[0..1].join(' ')
```

File: HW03/yuliya_sheremet/yuliya_sheremet_hw_03_t_03.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    if logs.scan('Calling core').length > 1
+      if_include(array_of_strings)
+      seconds = array_of_strings
+      check_and_print(seconds)
+    else
+      '0'
+    end
+  end
+
+  def print_a(seconds)
```
Comment (vladhilko) [inline HW03/yuliya_sheremet/yuliya_sheremet_hw_03_t_03.rb:24]: Can you give a clear name for `print_a` and `print_b` ?

File: HW03/yuliya_sheremet/yuliya_sheremet_hw_03_t_03.rb [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    arr = [] << (Time.parse(seconds[1]) - Time.parse(seconds[0])).to_s
+    arr << (Time.parse(seconds[2]) - Time.parse(seconds[1])).to_s
+  end
+
+  def check_and_print(seconds)
+    if seconds.size <= 2
+      print_a(seconds)
+    elsif seconds.size > 2
+      print_b(seconds)
+    end
```
Comment (vladhilko) [inline HW03/yuliya_sheremet/yuliya_sheremet_hw_03_t_03.rb:39]: we can use ternary operator here 
```suggestion
    seconds.size <= 2 ? print_a(seconds) :  print_b(seconds)
```

File: HW03/yuliya_sheremet/yuliya_sheremet_hw_03_t_03.rb [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  def task3(log)
+    array_of_strings = log.split(/\n/)
+    cases_for_output(log, array_of_strings)
+  end
+
+  def cases_for_output(logs, array_of_strings)
+    if logs.scan('Calling core').length > 1
+      if_include(array_of_strings)
+      seconds = array_of_strings
+      check_and_print(seconds)
```
Comment (vladhilko) [inline HW03/yuliya_sheremet/yuliya_sheremet_hw_03_t_03.rb:18]: we usually don't use `and` for method name, looks like `check` is not necessary here
```suggestion
      print(seconds)
```

**Comment by Julittaes** [comment]
Thank you for important remarks, I've fixed everything

**Pull request merged**